### PR TITLE
Make tuples with different arguments different types

### LINKED ISSUE #173: Got message "Hash code changed when merging AnalysisValues" with debug build
I wonder if it is an indication of some internal error potentially corrupting analysis?

It happens when I try to analyze installed tensorflow package. (e.g. pip install tensorflow-gpu, used version 1.10, then analyze site-packages/tensorflow).

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Fixes #173

Tuple protocols are compared by name which makes tuples with different types the same, which causes trouble in analysis hash set when it merges types of function parameters in overloads.

### CODE REVIEW

File: src/Analysis/Engine/Impl/Values/Protocols.cs [hunk 1]
```diff
@@ -415,7 +415,7 @@ class TupleProtocol : IterableProtocol {
             return AnalysisSet.UnionAll(constants.Select(GetItem));
         }
 
-        public override string Name => "tuple";
+        public override string Name => "tuple[{0}]".FormatInvariant(string.Join(", ", _values.Select(v => v.GetShortDescriptions())));
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/Values/Protocols.cs]: We can save it `ctor` to avoid multiple string creation.

**Comment by lostmsu** [comment]
Is short description unique enough? Should not `FullyQualifiedName` be used?

**Comment by MikhailArkhipov** [comment]
@lostmsu - sure, why not

**Comment by lostmsu** [comment]
@MikhailArkhipov FYI, just pulled this PR locally. Whatever issue I had with the assert is gone.

**Pull request merged**